volatility/framework/plugins/windows/registry/hivelist.py

```diff
@@ -8,10 +8,28 @@
 from volatility.framework.configuration import requirements
 from volatility.framework.layers import registry
 from volatility.framework.renderers import format_hints
+from volatility.plugins.windows.registry import hivescan
 
 vollog = logging.getLogger(__name__)
 
 
+class HiveGenerator():
+    """Walks the registry HiveList linked list in a given direction and stores an invalid offset
+    if it's unable to fully walk the list"""
+
+    def __init__(self, cmhive, forward = True):
+        self.cmhive = cmhive
+        self.forward = forward
```

Comment:
What happens if these are changed after construction?  I know it's a little long winded, but I'd prefer they were made into properties so that they're read only.  So essentially, `self._forward = forward` and then a forward property that simply returns `self._forward` for each of these variables.

Reply:
I started making properties for those, but really `invalid` is the only one that needs to be read publicly, so I made the other private and just a property for invalid.  I think that's what you were getting at, but let me know if I misunderstood.